# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieJarSettings.cpp
# Commit: af27682c5e7f
# Full Hash: af27682c5e7f86ff17d6ed152c8010e763870bdc
# Author: ssengupta <ssengupta@mozilla.com>
# Date: 2020-05-13 03:44:05
# Regressor Bug: 1635399
# File Overlap Count: 1
# Description:
#   Bug 1635399 - Function PrincipalInfoToPrincipal now returns Result<nsCOMPtr<nsIPrincipal>, nsresult> r=ckerschb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73868
# ==============================================================================

diff -r 86b22b1a9f7e -r af27682c5e7f netwerk/cookie/CookieJarSettings.cpp
--- a/netwerk/cookie/CookieJarSettings.cpp	Tue May 12 23:55:42 2020 +0300
+++ b/netwerk/cookie/CookieJarSettings.cpp	Tue May 12 19:02:05 2020 +0000
@@ -16,6 +16,7 @@
 #include "mozilla/StaticPrefs_network.h"
 #include "mozilla/Unused.h"
 #include "nsGlobalWindowInner.h"
+#include "nsIPrincipal.h"
 #if defined(MOZ_THUNDERBIRD) || defined(MOZ_SUITE)
 #  include "nsIProtocolHandler.h"
 #endif
@@ -286,12 +287,13 @@
 
   CookiePermissionList list;
   for (const CookiePermissionData& data : aData.cookiePermissions()) {
-    nsCOMPtr<nsIPrincipal> principal =
-        PrincipalInfoToPrincipal(data.principalInfo());
-    if (NS_WARN_IF(!principal)) {
+    auto principalOrErr = PrincipalInfoToPrincipal(data.principalInfo());
+    if (NS_WARN_IF(principalOrErr.isErr())) {
       continue;
     }
 
+    nsCOMPtr<nsIPrincipal> principal = principalOrErr.unwrap();
+
     nsCOMPtr<nsIPermission> permission =
         Permission::Create(principal, NS_LITERAL_CSTRING("cookie"),
                            data.cookiePermission(), 0, 0, 0);
@@ -347,12 +349,12 @@
   PermissionComparator comparator;
 
   for (const CookiePermissionData& data : aData.cookiePermissions()) {
-    nsCOMPtr<nsIPrincipal> principal =
-        PrincipalInfoToPrincipal(data.principalInfo());
-    if (NS_WARN_IF(!principal)) {
+    auto principalOrErr = PrincipalInfoToPrincipal(data.principalInfo());
+    if (NS_WARN_IF(principalOrErr.isErr())) {
       continue;
     }
 
+    nsCOMPtr<nsIPrincipal> principal = principalOrErr.unwrap();
     nsCOMPtr<nsIPermission> permission =
         Permission::Create(principal, NS_LITERAL_CSTRING("cookie"),
                            data.cookiePermission(), 0, 0, 0);